pkg/planner/core/stats.go

```diff
@@ -710,3 +721,385 @@ func derivePathStatsAndTryHeuristics(ds *logicalop.DataSource) error {
+	for _, col := range orderingColumns {
+		// Skip if already in whereColIDs to avoid double-counting
+		if _, exists := whereColIDs[col.ID]; !exists {
+			orderingColIDs[col.ID] = struct{}{}
+			totalOrderingColumns++
+		}
+	}
+	// Calculate total coverage as a single table index
+	totalLocalRequiredCols := totalWhereColumns + totalOrderingColumns
+	// Calculate total coverage as a join table index
+	totalJoinRequiredCols := totalJoinColumns
+	if totalJoinRequiredCols > 0 {
+		totalJoinRequiredCols += totalWhereColumns
+	}
+	maxConsecutiveWhere, maxConsecutiveJoin := 0, 0
+
+	// maxIndexes allows a minimum number of indexes to be kept regardless of threshold.
+	// max/minToKeep only calculate index plans to keep in addition to table plans.
+	// This avoids extreme pruning when threshold is very low (or even zero).
+	const maxIndexes = 10
+	maxToKeep := max(maxIndexes, threshold)
+	minToKeep := max(1, min(maxIndexes, threshold))
+	// Prepare lists to hold indexes with different levels of coverage
+	perfectCoveringIndexes := make([]indexWithScore, 0, maxIndexes) // Indexes covering all required columns
+	preferredIndexes := make([]indexWithScore, 0, maxIndexes)       // "Next" best indexes with preferable coveraage
+	tablePaths := make([]*util.AccessPath, 0, 1)                    // Usually just one table path
+
+	lenPerfectCoveringIndexes, lenPreferredIndexes := 0, 0
+	hasSingleScan := false
+	for _, path := range paths {
+		if path.IsTablePath() {
+			tablePaths = append(tablePaths, path)
+			continue
+		}
+
+		// If we have forced paths, we shouldn't prune any paths.
+		// Hints processing should have already done the pruning.
+		// It means we may have entered this function due to a low threshold.
+		if path.Forced {
+			return paths
+		}
+
+		// Count how many WHERE, join and ordering columns are covered by this index
+		// Also track consecutive matches from the start (most valuable for index usage)
+		coveredWhereCount := 0
+		coveredOrderingCount := 0
+		coveredJoinCount := 0
+		consecutiveWhereCount := 0
+		consecutiveOrderingCount := 0
+		consecutiveJoinCount := 0
+
+		skipThisIndex := false
+
+		for i, idxCol := range path.FullIdxCols {
+			idxColID := idxCol.ID
+
+			// Check if this index column matches a WHERE column
+			if totalWhereColumns > 0 {
+				if _, found := whereColIDs[idxColID]; found {
+					coveredWhereCount++
+					// Track consecutive matches from start of index
+					if i == consecutiveWhereCount {
+						consecutiveWhereCount++
+					}
+					// Also increment consecutive join count if there has already been at least one
+					// consecutive join match, and the current index column position matches the
+					// sum of consecutive join and where matches.
+					if consecutiveJoinCount > 0 && i == consecutiveJoinCount+consecutiveWhereCount {
+						consecutiveJoinCount++
+					}
+					continue // Move to next index column - since a column can only appear once
```

Comment:
It seems this `continue` also skips the following process of handling join and ordering column, is this expected?